### ISSUE TITLE
hydra job logging is not compatible with wandb - by hand logger configuration works![Bug]

### ISSUE THREAD

**Issue by somedadaism** [facebookresearch/hydra#1553]
**Description**
I try to combine `hydra` and `wandb` in a common framework.
`hydra` configures my (python, not wandb) loggers in a manner I think is convenient and allows to flexible alter them.
However the logger configurations made by `hydra` are overwritten by the `wandb.init()` function.
The issue exists in public repos too:
https://github.com/ashleve/lightning-hydra-template 
is an example I checked.

**How to reproduce**
```
import hydra
import sys
import wandb
import logging
import warnings

log = logging.getLogger(__name__)
warnings.filterwarnings('ignore', category=UserWarning)


@hydra.main(config_path='../configs', config_name='benchmark_config.yaml')
def reproduce_bug(config):
    log.info('foo')
    wandb.init(project='Test',
               dir='..',
               reinit=True)
    log.info('bar')


if __name__ == '__main__':
    reproduce_bug()
"""
output: 
[2021-04-15 11:17:16,642][__main__][INFO] - foo

expected: 
[2021-04-15 11:17:16,642][__main__][INFO] - foo
[2021-04-15 11:17:16,642][__main__][INFO] - bar
```

**Environment**
Multiple, python3.8 on Mac, Ubuntu

Originally I thought this was an issue with wandb, however I have since experimented and learned that this works:

```
import sys
import logging
import wandb


def reproduce_bug():
    log = logging.getLogger()
    root = logging.getLogger()
    root.setLevel(logging.INFO)
    handler = logging.StreamHandler(sys.stdout)
    formatter = logging.Formatter(
        "[%(asctime)s][%(name)s][%(levelname)s] - %(message)s"
    )
    handler.setFormatter(formatter)
    root.addHandler(handler)

    log.info('foo')
    wandb.init(project='Test',
               dir='..',
               reinit=True)
    log.info('bar')

reproduce_bug()
output: 
[2021-04-15 11:17:16,642][__main__][INFO] - foo
[2021-04-15 11:17:16,642][__main__][INFO] - bar

expected: 
[2021-04-15 11:17:16,642][__main__][INFO] - foo
[2021-04-15 11:17:16,642][__main__][INFO] - bar
```
My understanding is that it is possible to configure job logging in a wandb compatible way but hydra does not do this.
I have raised a parallel issue on `wandb/client`: https://github.com/wandb/client/issues/2075
Hope this helps to resolve the conflict.
hydra version is 1.1.0dev3

**Comment by ashleve** [comment]
@somedadaism Hi, 
I believe this issue exists only when running wandb in thread mode (WANDB_START_METHOD=thread).
It has been discussed here: https://github.com/facebookresearch/hydra/issues/1486
For me it works correctly when the env variable WANDB_START_METHOD=thread is removed, but then hydra sweeps with wandb will crash after 1 job.

**Comment by omry** [comment]
Thanks for reporting @somedadaism, as @ashleve  pointed out - this has already been discussed.
I suggest you iron this out with wandb, I am not sure why they are interfering with the logging and how to prevent them from doing it.
If you want to allow them to full control the logging, you can completely disable logging configuration in Hydra (see the last paragraph [here](https://hydra.cc/docs/next/tutorials/basic/running_your_app/logging)).

**Comment by somedadaism** [comment]
@omry Thank you for pointing out how to disable hydra logging.
I have read the documentation extensively and I prefer to use hydra logging over wandb logging.
@ashleve I have disabled the `WANDB_START_METHOD=thread` variable and my fork of your code runs hydra sweeps without problems. Apparently the issue has been resolved by W&B since.

**Comment by omry** [comment]
Great, I am closing this.
I suggest that you explain the situation to wandb based on the knowledge here and in #1486.
Hopefully they can fix their logging configuration to not interfere with existing loggers.

**Comment by somedadaism** [comment]
@omry I have made them aware of this issue - they say they are tracking it internally.

**Comment by vanpelt** [comment]
Thanks for this everyone.  We're looking into the root cause and are tracking it [here](https://github.com/wandb/client/issues/2075)

**Comment by omry** [comment]
> Thanks for this everyone. We're looking into the root cause and are tracking it [here](https://github.com/wandb/client/issues/2075)

Thanks @vanpelt (Nice to see you here :) ).

**Comment by omry** [comment]
@vanpelt, by the way:
Hydra 1.1 (getting close to release candidate) have a new callback mechanism that is not yet documented - that will allow nicer integration with experimentation tracking frameworks like wandb.

Essentially, it will allow the app to configure callbacks to be executed in various points in its life-cycle. in this case it will be useful to interface with job start/end and sweep start/end. The callback will be able to initialize wandb automatically at the right moment.

The design doc is [here](https://docs.google.com/document/d/1EDO0X3un9iLf65oYEkKgQ48qIAtYu3UYc64uSDiynKo/edit).
(Note that it may be a bit hard to apply because it's not meant as an end user document).
We should have real documentation before 1.1 is released (#1526).
